src/app/clusters/scenes-server/scenes-server.cpp

```diff
@@ -0,0 +1,993 @@
+    if (0 != groupID && !groupProvider->HasEndpoint(fabricIdx, groupID, endpointID))
+    {
+        return CHIP_IM_GLOBAL_STATUS(InvalidCommand);
+    }
+
+    // Scene Table interface data
+    SceneTableEntry scene(SceneStorageId(sceneID, groupID));
+
+    VerifyOrReturnError(nullptr != sceneTable, CHIP_ERROR_INTERNAL);
+    CHIP_ERROR err = sceneTable->GetSceneTableEntry(fabricIdx, scene.mStorageId, scene);
+    if (CHIP_NO_ERROR != err && CHIP_ERROR_NOT_FOUND != err)
+    {
+        return err;
+    }
+
+    if (CHIP_ERROR_NOT_FOUND == err)
+    {
+        scene.mStorageData.SetName(CharSpan());
+        scene.mStorageData.mSceneTransitionTimeMs = 0;
+    }
+    else
+    {
+        scene.mStorageData.mExtensionFieldSets.Clear();
+    }
+
+    // Gets the EFS
+    ReturnErrorOnFailure(sceneTable->SceneSaveEFS(scene));
+    // Insert in Scene Table
+    ReturnErrorOnFailure(sceneTable->SetSceneTableEntry(fabricIdx, scene));
+
+    // Update size attributes
+    uint8_t scene_count = 0;
+    ReturnErrorOnFailure(sceneTable->GetFabricSceneCount(fabricIdx, scene_count));
+
+    ReturnErrorOnFailure(StatusIB(ToInteractionModelStatus(Attributes::SceneCount::Set(endpointID, scene_count))).ToChipError());
+
+    uint8_t capacity = 0;
+    ReturnErrorOnFailure(sceneTable->GetRemainingCapacity(fabricIdx, capacity));
+
+    ReturnErrorOnFailure(
+        StatusIB(ToInteractionModelStatus(Attributes::RemainingCapacity::Set(endpointID, capacity))).ToChipError());
+
+    ReturnErrorOnFailure(StatusIB(ToInteractionModelStatus(Attributes::CurrentScene::Set(endpointID, sceneID))).ToChipError());
+
+    ReturnErrorOnFailure(StatusIB(ToInteractionModelStatus(Attributes::CurrentGroup::Set(endpointID, groupID))).ToChipError());
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR RecallSceneParse(const FabricIndex & fabricIdx, const EndpointId & endpointID, const GroupId & groupID,
+                            const SceneId & sceneID, const Optional<DataModel::Nullable<uint16_t>> & transitionTime,
+                            GroupDataProvider * groupProvider)
+{
+    uint16_t endpoinTableSize = 0;
+    ReturnErrorOnFailure(
+        StatusIB(ToInteractionModelStatus(Attributes::SceneTableSize::Get(endpointID, &endpoinTableSize))).ToChipError());
+
+    // Get Scene Table Instance
+    scenes::SceneTable<scenes::ExtensionFieldSetsImpl> * sceneTable = chip::scenes::GetSceneTableImpl(endpointID, endpoinTableSize);
+
+    // Verify Endpoint in group
+    VerifyOrReturnError(nullptr != groupProvider, CHIP_ERROR_INTERNAL);
+    if (0 != groupID && !groupProvider->HasEndpoint(fabricIdx, groupID, endpointID))
+    {
+        return CHIP_IM_GLOBAL_STATUS(InvalidCommand);
+    }
+
+    // Scene Table interface data
+    SceneTableEntry scene(SceneStorageId(sceneID, groupID));
+
+    VerifyOrReturnError(nullptr != sceneTable, CHIP_ERROR_INTERNAL);
+    ReturnErrorOnFailure(sceneTable->GetSceneTableEntry(fabricIdx, scene.mStorageId, scene));
```

Comment:
StatusIB does not convert CHIP_ERROR_NOT_FOUND into an IM status...  We need to do that here.